addons/web/static/src/model/relational_model/record.js

```diff
@@ -489,7 +494,7 @@ export class Record extends DataPoint {
         for (const fieldName in data) {
             const value = data[fieldName];
             const field = this.fields[fieldName];
-            if (field.relatedPropertyField) {
+            if (!field || field.relatedPropertyField) {
```

Comment:
This is wrong. This should not happen, so we don't want to handle such a case. I don't know why you reach the situation where a field in data doesn't exist, but this is what needs to be fixed.